- [ ] TODO resolve issue with segmentation model

<details><summary>CI Benchmark error</summary>
<p>

```
TensorFlow GraphDef: starting export with tensorflow 2.16.1...
TensorFlow GraphDef: export success ✅ 3.9s, saved as 'path with spaces/yolov8n-seg.pb' (13.1 MB)

Export complete (31.4s)
Results saved to /home/runner/work/ultralytics/ultralytics/path with spaces
Predict:         yolo predict task=segment model=path with spaces/yolov8n-seg.pb imgsz=160  data=coco.yaml
Validate:        yolo val task=segment model=path with spaces/yolov8n-seg.pb imgsz=160 data=coco.yaml  
Visualize:       https://netron.app/
Loading path with spaces/yolov8n-seg.pb for TensorFlow GraphDef inference...

Traceback (most recent call last):
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/utils/benchmarks.py", line 126, in benchmark
    exported_model.predict(ASSETS / "bus.jpg", imgsz=imgsz, device=device, half=half)
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/model.py", line 453, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/predictor.py", line 168, in __call__
    return list(self.stream_inference(source, model, *args, **kwargs))  # merge list of Result into one
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 35, in generator_context
    response = gen.send(None)
               ^^^^^^^^^^^^^^
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/predictor.py", line 255, in stream_inference
    self.results = self.postprocess(preds, im, im0s)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/models/yolo/segment/predict.py", line 30, in postprocess
    p = ops.non_max_suppression(
        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/utils/ops.py", line [235](https://github.com/ultralytics/ultralytics/actions/runs/9387491195/job/25850532202?pr=13389#step:8:236), in non_max_suppression
    output = [torch.zeros((0, 6 + nm), device=prediction.device)] * bs
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Trying to create tensor with negative dimension -882: [0, -882]
```

</p>
</details>

--- a/ultralytics/nn/autobackend.py
+++ b/ultralytics/nn/autobackend.py
@@ -567,7 +567,7 @@ class AutoBackend(nn.Module):
                     y = [y]
             elif self.pb:  # GraphDef
                 y = self.frozen_func(x=self.tf.constant(im))
-                if self.task == 'segment' or (len(y) == 2 and len(self.names) == 999):  # segments and names not defined
+                if self.task == "segment" or (len(y) == 2 and len(self.names) == 999):  # segments and names not defined
                     ip, ib = (0, 1) if len(y[0].shape) == 4 else (1, 0)  # index of protos, boxes
                     nc = y[ib].shape[1] - y[ip].shape[3] - 4  # y = (1, 160, 160, 32), (1, 116, 8400)
                     self.names = {i: f"class{i}" for i in range(nc)}

--- a/ultralytics/nn/autobackend.py
+++ b/ultralytics/nn/autobackend.py
@@ -567,7 +567,7 @@ class AutoBackend(nn.Module):
                     y = [y]
             elif self.pb:  # GraphDef
                 y = self.frozen_func(x=self.tf.constant(im))
-                if len(y) == 2 and len(self.names) == 999:  # segments and names not defined
+                if self.task == 'segment' or (len(y) == 2 and len(self.names) == 999):  # segments and names not defined
                     ip, ib = (0, 1) if len(y[0].shape) == 4 else (1, 0)  # index of protos, boxes
                     nc = y[ib].shape[1] - y[ip].shape[3] - 4  # y = (1, 160, 160, 32), (1, 116, 8400)
                     self.names = {i: f"class{i}" for i in range(nc)}

--- a/ultralytics/nn/autobackend.py
+++ b/ultralytics/nn/autobackend.py
@@ -321,7 +321,7 @@ class AutoBackend(nn.Module):
                 gd.ParseFromString(f.read())
             frozen_func = wrap_frozen_graph(gd, inputs="x:0", outputs=gd_outputs(gd))
             try:
-                metadata = next(Path(w).resolve().parent.rglob("metadata.yaml"))
+                metadata = next(Path(w).resolve().parent.rglob(f"{Path(w).stem}_saved_model*/metadata.yaml"))
             except StopIteration:
                 pass  # no metadata file found
 

